src/referenceHandling.ts

```diff
@@ -0,0 +1,172 @@
+// (C) 2007-2019 GoodData Corporation
+import { VisualizationObject } from '@gooddata/typings';
+import { cloneDeep, get, pickBy, set } from 'lodash';
+import { v4 as uuid } from 'uuid';
+
+import {
+    IReferenceProperties,
+    IColorMappingReferenceProperty,
+    IColorMappingUriProperty,
+    IUriProperties
+} from './interfaces';
+
+type IVisualizationObject = VisualizationObject.IVisualizationObject;
+type IReferenceItems = VisualizationObject.IReferenceItems;
+
+/**
+ * Constants
+ */
+const CONTROLS_PROPERTY_PATH = ['controls'];
+const COLOR_MAPPING_PROPERTY_PATH = [...CONTROLS_PROPERTY_PATH, 'colorMapping'];
+
+/**
+ * Helpers
+ */
+const getReferenceValue = (id: string, references: IReferenceItems) => references[id];
+const getReferenceId = (value: string, references: IReferenceItems) =>
+    Object.keys(references).find(id => references[id] === value);
+
+const defaultIdGenerator = () => uuid().replace(/-/g, '');
+
+const createConverter = <TInProperties, TOutProperties>(
+    conversionBody:
```

Comment:
This is not very readable, also `conversionBody` is a function which is a bit confusing

Reply:
Renamed to more clearly convey the meaning.